sdk-setup/src/mb2

```diff
@@ -1032,8 +1032,13 @@ maybe_verify_target_dependencies() {
             filter_repo "$exclude_source_name" "$OPT_OUTPUTDIR" "$filtered_outputdir"
             local args=(--"$OPT_SEARCH_OUTPUTDIR_VERBOSITY" --non-interactive \
                 --plus-repo "$filtered_outputdir" --no-gpg-checks)
-            sb2 -t "$OPT_TARGET" -m sdk-install -R zypper "${args[@]}" dup --from '~plus-repo-1'
+            # Install first, then dup to ensure we really use packages from
+            # the output directory as the repos may contain newer packages and
+            # those would be pulled in preferably.  This happens not only when
+            # working with older sources but also when a package hardcodes
+            # version in its spec and the --fix-version option is not active.
```

Comment:
Now this seems so logical that I'm starting to wonder if there was some reason (that I'm not aware of and you have already forgotten) why dup was originally performed before install?

Reply:
Yeah, seems logical but it feels more natural/safe to dup first and install new packages later - this is what you would normally do to your system, right? In theory, ordered install-first it could fail temporarily and there is the possibility to get a package pulled from a regular repo first and downgraded to the local built version in a second step - harmless but slightly annoying.